components/Table.vue

```diff
@@ -187,8 +171,18 @@
         },
 
         watch: {
-            searchParams() {
-                this.search();
+            searchParams: {
+                handler: function(n) {
+                    if (n.selectedCol !== this.defaultParams.selectedCol) {
```

Comment:
Again you can probably just loop through the keys in the `n` variable to check what actually exists.